parser/gotest/gotest.go

```diff
@@ -41,7 +41,10 @@ var (
 		`(?:\s+(\[[^\]]+\]))?` +
 		// 6: coverage percentage (optional)
 		// 7: coverage package list (optional)
-		`(?:\s+coverage:\s+(\d+\.\d+)%\sof\sstatements(?:\sin\s(.+))?)?$`)
+		`(?:\s+coverage:\s+(?:\[no\sstatements\]|(\d+\.\d+)%\sof\sstatements(?:\sin\s(.+))?))?` +
+		// 8: [additional status message] (optional)
+		`(?:\s+(\[[^\]]+\]))?` +
```

Comment:
I didn't see such case yet. But I think it doesn't hurt to keep both?

Reply:
I also haven't seen such a case, which means it won't be covered by a test. We'd also have to keep maintaining it going forward and it might make future changes to this already complicated regular expression more difficult. Let's remove it.

Reply:
I am not very sure if I understand the problem. What do you mean by "remove group 5"? You mean delete the capture group 5 from the regular expression? Then won't we failed to match cases when only group 5 is there but group 8 is not?

Reply:
Yes, that's what I meant. Groups 5 and 8 try to capture the same thing, a string enclosed by `[` and `]`. The only thing that influences which of these groups contains the actual string is whether or not coverage information was present. But since the capture group for coverage information is optional, we could just let group 8 capture everything.

Reply:
Done